spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/LoadBalancerRetryPolicy.java

```diff
@@ -32,27 +32,71 @@
 	 * @param context the context for the retry operation
 	 * @return true to retry on the same service instance
 	 */
+	@Deprecated
 	boolean canRetrySameServiceInstance(LoadBalancerRetryContext context);
 
+	/**
+	 * Return <code>true</code> to retry on the same service instance.
+	 * @param serviceId the serviceId for the retry operation
+	 * @param context the context for the retry operation
+	 * @return true to retry on the same service instance
+	 */
+	default boolean canRetrySameServiceInstance(String serviceId, LoadBalancerRetryContext context) {
+		return canRetrySameServiceInstance(context);
+	}
+
 	/**
 	 * Return <code>true</code> to retry on the next service instance.
 	 * @param context the context for the retry operation
 	 * @return true to retry on the same service instance
 	 */
+	@Deprecated
 	boolean canRetryNextServiceInstance(LoadBalancerRetryContext context);
 
+	/**
+	 * Return <code>true</code> to retry on the next service instance.
+	 * @param serviceId the serviceId for the retry operation
+	 * @param context the context for the retry operation
+	 * @return true to retry on the same service instance
+	 */
+	default boolean canRetryNextServiceInstance(String serviceId, LoadBalancerRetryContext context) {
+		return canRetryNextServiceInstance(context);
+	}
+
 	/**
 	 * Return <code>true</code> to retry on the provided HTTP status code.
 	 * @param statusCode the HTTP status code
 	 * @return true to retry on the provided HTTP status code
 	 */
+	@Deprecated
 	boolean retryableStatusCode(int statusCode);
 
+	/**
+	 * Return <code>true</code> to retry on the provided HTTP status code.
+	 * @param serviceId the serviceId for the retry operation
+	 * @param statusCode the HTTP status code
+	 * @return true to retry on the provided HTTP status code
+	 */
+	default boolean retryableStatusCode(String serviceId, int statusCode) {
+		return retryableStatusCode(statusCode);
+	}
+
 	/**
 	 * Return <code>true</code> to retry on the provided HTTP method.
 	 * @param method the HTTP request method
 	 * @return true to retry on the provided HTTP method
 	 */
+	@Deprecated
```

Comment:
Please add javadoc with `@deprecated` tag indicating the suggested replacement.

Reply:
Added